reactor-core/src/main/java/reactor/core/publisher/FluxWindowTimeout.java

```diff
@@ -414,14 +415,23 @@ boolean enter() {
 			return WIP.getAndIncrement(this) == 0;
 		}
 
-		static final class ConsumerIndexHolder implements Runnable {
+		static final class ConsumerIndexHolder implements Scheduler.ContextRunnable {
 
 			final long                       index;
 			final WindowTimeoutSubscriber<?> parent;
+			final Context                    context;
 
-			ConsumerIndexHolder(long index, WindowTimeoutSubscriber<?> parent) {
+			ConsumerIndexHolder(long index,
+					WindowTimeoutSubscriber<?> parent,
+					Context context) {
 				this.index = index;
 				this.parent = parent;
+				this.context = context;
```

Comment:
as said above, can be avoided. `currentContext()` may return `this.parent.currentContext()`